File: backend/bikePartsHub/src/main/java/com/bphTeam/bikePartsHub/entity/Bike.java
package com.bphTeam.bikePartsHub.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Bike {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "bike_id")
    private Long bikeId;

    private String type;
    private String model;
    private String version;
    private String manufacture;

    @OneToMany(mappedBy = "bikes")
    private Set<ProductAttribute> productAttributes;
}

File: frontend/BikePartsHubApp/src/app/core/models/interface/Bike.ts
export interface Bike {
  bikeId: number;
  type: string;
  model: string;
  version: string;
  manufacture: string;
}

export interface BikeSave {
  type: string;
  model: string;
  version: string;
  manufacture: string;
}


export interface BikeGet{
  bikeId:number
  type: string;
  model: string;
  version: string;
  manufacture: string;
}

export interface BikeUpdate {
  type: string;
  model: string;
  version: string;
  manufacture: string;
}

